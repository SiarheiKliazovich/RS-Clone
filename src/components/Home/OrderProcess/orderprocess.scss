@import "../../../sass/mixins";
@import "../../../sass/variables";



.banner-top {
  background-image: url(../../../assets/images/jpg/BannerTie.jpg);
  background-repeat: no-repeat;
  height: 150px;
  border-top: 3px solid #ffffff;
  border-bottom: 3px solid #ffffff;
}

.order-process__banner {
  margin-top: 0px;
  margin-bottom: 0px;
  background-color: #dfe3e6 ;
}

.title-block {
  text-align: center;
  padding: 10px 30px;
}

.order-title {
  font-weight: 200;
  color: #898989;
  font-size: 52px;
  padding-bottom: 15px;
}

.strong-title {
  font-weight: 700;
  color: #646464;
  font-size: $font-size-xxxxxlg;
  padding-left: 15px;
}

.order-title-text {
  font-family: $font-family-base;
  font-size: $font-size-llg;
}


.order-instruction-block {
  display: flex;
  justify-content: space-around;
  padding: 30px 0px;

  @include media-small-mobile {
    flex-direction: column;
  }
}

.image-order__wrapper {
  text-align: center;
}

.text-order__wrapper {
  display: flex;
  flex-direction: column;
}

.instruction-block__text {
  font-family: $font-family-base;
  font-size: $font-size-llg;
}

.strong-text {
  font-weight: 700;
  color: #646464;
  font-size: $font-size-llg;
}

.image-order {
  width: 192px;
  display: inline-block;
  background-repeat: no-repeat;
  height: 212px;
}

.first {
  background-image: url(https://cdn-bhllc.nitrocdn.com/tzrfwdCzhtOBUBnOQehAKoYljbaDwzgQ/assets/static/optimized/rev-c439613/wp-content/uploads/2019/11/0000169-Tie.png);
}

.second {
  background-image: url(https://cdn-bhllc.nitrocdn.com/tzrfwdCzhtOBUBnOQehAKoYljbaDwzgQ/assets/static/optimized/rev-c439613/wp-content/uploads/2019/11/0000263-colour-pallette.png);
}

.third {
  background-image: url(https://cdn-bhllc.nitrocdn.com/tzrfwdCzhtOBUBnOQehAKoYljbaDwzgQ/assets/static/optimized/rev-c439613/wp-content/uploads/2019/11/0000176-finish.png);
}



