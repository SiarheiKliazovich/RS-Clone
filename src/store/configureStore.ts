/* eslint-disable @typescript-eslint/ban-ts-comment */
import { applyMiddleware, createStore } from "redux";
import thunkMiddleware from "redux-thunk";

import loggerMiddleware from "./logger";
import rootReducer from "../reducers";

const configureStore = () => {
  const middlewares = [loggerMiddleware, thunkMiddleware];
  const middlewareEnhancer = applyMiddleware(...middlewares);

  const store = createStore(rootReducer, middlewareEnhancer);

  // @ts-ignore:next-line
  if (process.env.NODE_ENV !== "production" && module.hot) {
    // @ts-ignore:next-line
    module.hot.accept("./../reducers", () => store.replaceReducer(rootReducer));
  }

  return store;
};

export default configureStore;
