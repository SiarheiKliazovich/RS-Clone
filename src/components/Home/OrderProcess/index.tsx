import { FunctionComponent } from "react";
import { useSelector } from "react-redux";
import "./orderprocess.scss";
import orderprocess from "../../../data/orderprocess";
import Hover from "../../Hover";
import { Link } from "react-router-dom";
import { nightTheme } from "../../../data/constants";
import { OrderProcessType } from "../../../types/home";
import { ILangReducer } from "../../../interfaces/langReducer";

const OrderProcess: FunctionComponent<OrderProcessType> = ({
  accentColor,
  isNavbarNightMode,
}: OrderProcessType) => {
  const { lang } = useSelector((state: ILangReducer) => state.langReducer);
  const list = orderprocess.find((c) => c.lang === lang)!;
  const backgroundColor = nightTheme.background.element;

  return (
    <>
      <div className="order-process__wrapper">
        <div className="banner-top"></div>
        <div className="order-process__banner">
          <div className="title-block">
            <h2 className="order-title">{list.data.title}</h2>
            <p className="order-title-text">{list.data.text}</p>
          </div>
          <div className="order-instruction-wrapper">
            <div className="order-instruction-block">
              <div className="image-order__wrapper">
                <span className="image-order first"></span>
                <div className="text-order__wrapper">
                  <p className="instruction-block__text">{list.data.choose}</p>
                </div>
              </div>
              <div className="image-order__wrapper">
                <span className="image-order second"></span>
                <div className="text-order__wrapper">
                  <p className="instruction-block__text">{list.data.apply}</p>
                </div>
              </div>
              <div className="image-order__wrapper">
                <span className="image-order third"></span>

                <div className="text-order__wrapper">
                  <p className="instruction-block__text">{list.data.finish}</p>
                </div>
              </div>
            </div>
          </div>
          <div className="button-link-wrapper">
            <Hover>
              <button
                className="button-link"
                style={{
                  backgroundColor: isNavbarNightMode
                    ? backgroundColor
                    : accentColor.static,
                }}
              >
                <Link className="btn-video-link" to="/configurator">
                  {list.data.button}
                </Link>
              </button>
            </Hover>
          </div>
        </div>
      </div>
    </>
  );
};
export default OrderProcess;
