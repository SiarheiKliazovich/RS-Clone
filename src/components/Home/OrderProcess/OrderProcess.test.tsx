import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter as Router } from "react-router-dom";
import configureStore from "redux-mock-store";
import OrderProcess from ".";

const mockStore = configureStore([]);

describe("OrderProcess", () => {
  // eslint-disable-next-line
  let store: any;

  beforeEach(() => {
    store = mockStore({
      langReducer: {
        lang: "en",
      },
      appInterface: {
        accentColor: {
          static: "#000",
          hover: "#fff",
        },
      },
    });
  });

  test("renders correctly", () => {
    const { container } = render(
      <Provider store={store}>
        <Router>
          <OrderProcess
            accentColor={{ static: "#fff", hover: "#000" }}
            isNavbarNightMode={false}
          />
        </Router>
      </Provider>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
