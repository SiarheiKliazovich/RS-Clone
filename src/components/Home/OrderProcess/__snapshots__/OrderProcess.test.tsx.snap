// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`OrderProcess renders correctly 1`] = `
<div
  class="order-process__wrapper"
>
  <div
    class="banner-top"
  />
  <div
    class="order-process__banner"
  >
    <div
      class="title-block"
    >
      <h2
        class="order-title"
      >
        Order Process
      </h2>
      <p
        class="order-title-text"
      >
        We have made our design capabilities available to you, the real designer so that you can customise your own tie, scarf and even packaging â€“ enjoy!
      </p>
    </div>
    <div
      class="order-instruction-wrapper"
    >
      <div
        class="order-instruction-block"
      >
        <div
          class="image-order__wrapper"
        >
          <span
            class="image-order first"
          />
          <div
            class="text-order__wrapper"
          >
            <p
              class="instruction-block__text"
            >
              CHOOSE CANVAS
            </p>
          </div>
        </div>
        <div
          class="image-order__wrapper"
        >
          <span
            class="image-order second"
          />
          <div
            class="text-order__wrapper"
          >
            <p
              class="instruction-block__text"
            >
              APPLY COLOURS
            </p>
          </div>
        </div>
        <div
          class="image-order__wrapper"
        >
          <span
            class="image-order third"
          />
          <div
            class="text-order__wrapper"
          >
            <p
              class="instruction-block__text"
            >
              FINISH YOUR CREATION
            </p>
          </div>
        </div>
      </div>
    </div>
    <div
      class="button-link-wrapper"
    >
      <button
        class="button-link"
        style="background-color: rgb(0, 0, 0);"
      >
        <a
          class="btn-video-link"
          href="/configurator"
        >
          GO TO CONFIGURATOR
        </a>
      </button>
    </div>
  </div>
</div>
`;
